src/form/dragDropArea/index.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react'
+import { makeStyles } from '@material-ui/core'
+
+const useStyles = makeStyles({
+  root: {
+    height: '100vh',
+    width: '100%',
+  },
+})
+
+interface IDragDropAreaProps {
+  overlay: React.ReactElement
+  children: React.ReactElement
+  onFilesDrop: (files: File[]) => void
+}
+
+const DragDropArea = ({ overlay, children, onFilesDrop }: IDragDropAreaProps) => {
+  const classes = useStyles()
+  const [dragActive, setDragActive] = React.useState(0)
+
+  const onDragEnter = React.useCallback((event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault()
+    setDragActive(prev => prev + 1)
+  }, [])
```

Comment:
```suggestion
  }, [setDragActive])
```

Reply:
Not really sure if this is needed, the linter seems to think no. Looking at react docs https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies on the `ProductPage` example they don't include the state setter in the list of dependencies.

Reply:
That is true. The lint doesn't protest if you put it there either.